app/models/va_facility.rb

```diff
@@ -49,29 +46,25 @@ def self.find_by(id:)
     prefix, station = id.split('_')
     adapter = client_adapter(ID_PREFIXES[prefix])
     return nil unless adapter
-    request = adapter.find_by(id: station)
-    responses = multi_client.run([request])
-    adapter.class&.from_gis(responses.first.first) unless responses.first.blank?
+    adapter.get(station)
   end
 
   def self.service_whitelist(prefix)
-    client_adapter(prefix)&.service_whitelist
-  end
-
-  def self.multi_client
-    @mc ||= Facilities::MultiClient.new
+    client_adapter(prefix)&.adapter&.service_whitelist
   end
 
   def self.client_adapter(prefix)
-    @client_map ||= init_adapters
+    @lock.synchronize do
+      @client_map ||= init_adapters
+    end
     @client_map[prefix]
   end
 
   def self.init_adapters
     {
-      HEALTH => VHAFacilityAdapter.new,
-      CEMETERY => NCAFacilityAdapter.new,
-      BENEFITS => VBAFacilityAdapter.new
+      HEALTH => Facilities::LocalStore.new(ENV['VHA_MAPSERVER_URL'], nil, VHAFacilityAdapter),
+      CEMETERY => Facilities::LocalStore.new(ENV['NCA_MAPSERVER_URL'], nil, NCAFacilityAdapter),
```

Comment:
Chances are only one or more services might be able to be cached so was thinking having more separation of concerns might be more readable, just cache the method that fetches based on a uri for the key or something only if successful.

In which case the separation into distinct classes might make it more readable which of the clients need this. You could also then have custom middleware for each, because it seems as though maybe some of the clients handle errors one way and others handle them another, etc. It would be more composable and readable I think... But again, it's not even a minor gripe. 

The mutexes still seem weird to me, so I'm wondering if maybe having the 1 client to rule 3 others would kind of speak to the need of using mutex better too. But again, just a nitpick here. Something to think about in future refactor maybe.